.github/workflows/build-test-osx-x86.yaml

```diff
@@ -1,29 +1,53 @@
 # This workflow builds and tests the semgrep-core binary for macOS X86
 # and generates osx-wheel for pypi.
 
+# This workflow will cache the ~/.opam directory, which is not symmetic with our other architecture's build processes.
+# This runs on GHA-hosted runners, so without caching, it runs very slowly. The Linux build uses a container to bring in the
+# required dependencies, and the M1 build runs on a self-hosted machine where caching isn't necessary.
+# To update to a new version of OCaml, we can switch out `OPAM_SWITCH_NAME` var which will update the cache key, and lead to a cache miss
+# on the new builds.
+
 # coupling: if you modify this file, modify also build-test-osx-m1.yaml
 name: build-test-osx-x86
 
 on:
   workflow_dispatch:
+    inputs:
+      use-cache:
+        description: "Use Opam Cache - uncheck the box to disable use of the opam cache, meaning a long-running but completely from-scratch build."
+        required: true
+        type: boolean
+        default: true
   workflow_call:
+    inputs:
+      use-cache:
+        description: "Use Opam Cache - uncheck the box to disable use of the opam cache, meaning a long-running but completely from-scratch build."
+        required: false
+        type: boolean
+        default: true
 
 jobs:
   build-core-osx:
     name: Build the OSX binaries
-    runs-on: [self-hosted, macOS, X64]
+    runs-on: macos-12
+    env:
+      # This name is used in the cache key. If we update to a newer version of ocaml, we'll want to change the OPAM_SWITCH_NAME
+      # as well to avoid issues with caching.
+      OPAM_SWITCH_NAME: "4.14.0"
     steps:
-      # see the comment in build-test-osx-m1.yaml why we need this step
-      - name: Clean Checkout
-        run: |
-          rm -rf ./*
-          rm -f .gitmodules
-          rm -rf .git
       - uses: actions/checkout@v3
         with:
           submodules: true
+      # Note that this action handles both cache read and cache write.
+      - name: Cache Opam
+        id: cache-opam
+        uses: actions/cache@v3
+        if: ${{ inputs.use-cache }}
+        with:
+          path: ~/.opam
+          key: "${{ runner.os }}-${{ runner.arch }}-${{ env.OPAM_SWITCH_NAME }}-opam-deps"
       - run: |
-          ./scripts/osx-setup-for-release.sh
+          ./scripts/osx-setup-for-release.sh "${{ env.OPAM_SWITCH_NAME }}"
           opam exec -- make core
           mkdir -p artifacts
           cp ./bin/semgrep-core artifacts
```

Comment:
Please give a name to this step... It took me a while to figure out why semgrep was being built by the script `./scripts/osx-setup-for-release.sh`. This script doesn't build semgrep but it's the title that GHA assigns automatically to the step.

Reply:
I just did that.

Reply:
I think I'm going to split the step into installing dependencies and compiling semgrep since both are big.

Reply:
I did the split. I hope it works.

Reply:
It took 7 min, this time. Could be 5 min if we skipped `opam update`.